### DISCUSSION TITLE
How to put drop-shadows on all sides of a Polaroid?

### DISCUSSION THREAD

**Discussion by halloleo** [ImageMagick/ImageMagick#6362] [General] [answered]
I love the `-polaroid` operator, but would like to tweak things a bit when I layer multiple Polaroids. The top edges of the layered Polaroids are pretty indistinguishable:

<img width="660" alt="board_array" src="https://github.com/ImageMagick/ImageMagick/assets/1521852/9fc61e37-4bb5-4f13-bb6a-4f5224a6bf1e">

Therefore  I thought drop-shadows on _all_ sides, not only at the bottom edge would help here. But how can I achieve this?

**Comment by fmw42** [comment]
Add a shadow to each image before doing the polaroid.  See https://imagemagick.org/Usage/blur/#shadow

**Reply by halloleo** [reply]
Mmmmh, thanks for chiming in! 😄  Interesting idea, but when I do `-shadow ...` (for the upper shadow)  _before_ I do the `-polaroid` operator, the upper shadow is applied to the image(s) _without_ the border the Polaroid effect applied:

<img width="666" alt="board_array" src="https://github.com/ImageMagick/ImageMagick/assets/1521852/e0142659-03ea-4923-976d-15c05bce8fe9">

However I would like the additional upper shadow applied to the frame where `-polaroid` applied its lower shadow.

Is there a way to do the Polaroid operator manually? Then I could jump right in where the shadow is applied...

**Comment by snibgo** [comment, marked as answer]
@halloleo: What command(s) did you use? What is your source image? What is your platform?

`-polaroid` adds a border that is white, or uses your current `-bordercolor`, and then adds a shadow. You can add an extra shadow after that. `-polaroid` does not have a facility to turn off the shadow it adds.

For example, you can do this (Windows BAT syntax)
```
magick ^
  -caption toes.png ^
  toes.png ^
  -background Black ^
  -polaroid 10 +repage ^
  ( +clone ^
    -shadow 80x3+0+0 ^
  ) ^
  +swap ^
  -background None -layers merge +repage ^
  s.png
```
![s](https://github.com/ImageMagick/ImageMagick/assets/33812368/69f73d6d-ad9a-423f-a9c3-575e8af65ff5)

You can see what `-polaroid` does from the source code in https://github.com/ImageMagick/ImageMagick/blob/main/MagickCore/visual-effects.c , so you could replicate the operations at the command line, substituting your own shadow effect.

**Reply by halloleo** [reply]
Super cool! Applying the shadow _after_ the polaroid operator works pretty well:

<img width="666" alt="board_array" src="https://github.com/ImageMagick/ImageMagick/assets/1521852/e54eee1d-cf03-437e-b868-ce6411ba3b33">

Thanks heaps!
